jwt.js

```diff
@@ -176,11 +222,21 @@ function fastifyJwt (fastify, options, next) {
   function replySign (payload, options, next) {
     if (typeof options === 'function') {
       next = options
-      options = Object.assign({}, signOptions)
+      options = {}
     } // support no options
 
     if (!options) {
-      options = Object.assign({}, signOptions)
+      options = {}
+    }
+
+    if (options.sign) {
+      // New supported contract, options supports sign and can expand
```

Comment:
This interface is clearer now, but why is this new option needed?
What use case are we covering?

Reply:
The intention was to keep all the callable methods being exported to support the same a config object of the same schema. This would enable end-users to set the config at plugin registration or on-the-fly, using a consistent schema.

It doesn't change `replySign` as it is used now, but it does allow for new future functionality. If `replySign` is expanded to include other configurations, that are unrelated to the call of the `jwt.sign` function, with the above change it can support it.

@Eomm More clear? Happy to revert until the need arises, but I originally added it to keep all the "public" methods consistent in behavior around config.

E.g. Calling `replySign` with config, only the `config.sign` attribute will be passed to `jwt.sign`.

Reply:
I think it is.
I'm not a fan of this, we are suggesting to the user to write:

```
const config = {
  sign: {},
  decode: []
}

reply.jwtSign('foobar', config)
reply.jwtDecode('axaxa', config)

// instead of

reply.jwtSign('foobar', config.sign)
reply.jwtDecode('axaxa', config.decode)
```

I would like to see a real need for this usage, otherwise, it can become a headache to maintain all these multiple interfaces (maybe it is needed by the https://github.com/nearform/fastify-auth0-verify /? )

Apart from this API concern, the code is good, could you add the tests for the new decorator?

Reply:
@Eomm That's the idea, but I think it is worth seeing it used for `jwtVerify`.

```js
reply.jwtSign(myPayload, {
  sign: {
    issuer: 'foo.baz.baz'
  }
})

request.jwtVerify(myPayload, {
  verify: {
    issuer: 'foo.baz.baz'
  },
  sign: {
    issuer: 'foo.baz.baz'
  }
})
```

The adjustment I am advocating for is to allow for user-driven configuration of the `jwt` functions being wrapped.

- `request.jwtVerify` wraps `jwt.decode` and `jwt.verify`
- `request.sign` wraps `jwt.sign`
- `request.decode` wraps `jwt.decode`, but also uses the `extractToken` function attributed to the verify config.

The intention was by being explicit about the usage of the configuration, the code would be easier to use/understand. As a side effect, we would have a single "fastify-jwt" config schema, rather than varying per each method.

I will put together the PR for fastify-auth0-verify and link it to this momentarily.

The tests are up to date now also.

Reply:
Now I get it
Thanks for the clarification

We need to update the README accordingly, adding the decorator